### DISCUSSION TITLE
Troubleshooting "permission denied for database postgres"

### DISCUSSION THREAD

**Discussion by kjcsb1** [PostgREST/postgrest#3784] [Q&A] [answered]
PostgREST 12.2

I'm connecting PostgREST to a restored database from a dump created using pgdumpall. The source database has a PostgREST instance successfully connected. However PostgREST returns a `permission denied for database postgres` message. I feel like I'm missing something obvious. Any suggestions appreciated.

**Comment by laurenceisla** [comment, marked as answer]
> `permission denied for database postgres`

It looks to me that the user does not have connection permissions to the database. Perhaps it has to do with a `pg_dumpall` option (e.g. the `-x` option does not migrate privileges). Check if you you can connect to the database with `psql` and your "authenticator" user.

If not, try granting connection privileges and such:

```
GRANT CONNECT ON DATABASE postgres TO your-authenticator;
```

If that's so, check if your `pg_dumpall` is migrating the permissions correctly.

**Reply by kjcsb1** [reply]
Thanks very much for the clue. It was indeed a permissions issue for the role that I was using.
